### PR TITLE
Added HEIC support

### PR DESCRIPTION
**Pull request by tttdevelop**
OpenCV still does not support HEIC file format. This is a workaround to support HEIC files and convert to OpenCV format.

### CODE REVIEW

File: deepface/commons/functions.py [hunk 1]
```diff
@@ -70,7 +71,21 @@ def load_image(img):
 
     # ---------------------------
 
-    if base64_img is True:
+    filename, file_extension = os.path.splitext(img)
```
Comment (serengil) [inline deepface/commons/functions.py:74]: this logic should be done in `exact_image is not True` block

File: requirements.txt [hunk 2]
```diff
@@ -14,4 +14,5 @@ dlib>=19.20.0
 retina-face>=0.0.1
 mediapipe>=0.8.7.3
 fire>=0.4.0
-gunicorn>=20.1.0
\ No newline at end of file
+gunicorn>=20.1.0
+pyheif>=0.7.1
```
Comment (serengil) [inline requirements.txt:18]: requirements.txt is just showing dependencies. we are not using this in the installation of the library. this dependency should be added in setup.py.

**Comment by serengil** [comment]
thank you for your contribution.

i added some comments.

File: deepface/commons/functions.py (hunk 1 above)
Comment (tttdevelop) [reply deepface/commons/functions.py:74]: No problems. Before I update the code, is this the format you are looking for ?

```
    elif exact_image is not True:  # image path passed as input
        if os.path.isfile(img) is not True:
            raise ValueError(f"Confirm that {img} exists")

        if (file_extension.lower() == ".heic"):
            ...
        else:
            img = cv2.imread(img)

```

**Comment by bigcat88** [comment]
It may be easier to use [pillow-heif](https://github.com/bigcat88/pillow_heif) for this (code for the latest version 0.10.0):

``` python3
          img = np.array(pillow_heif.open_heif(img, bgr_mode=true))
```
Or use even a light version called pi-heif, because it seems that you only need decoding functionality.

**Comment by tttdevelop** [comment]
Interesting, I'll look at both, maybe pi-heif is a better option.

**Comment by Vincent-Stragier** [comment]
@tttdevelop,

[pi-heif](https://pypi.org/project/pi-heif/) looks powerful. Apparently, you can use it as a Pillow plugin. From the doc:

```python
from PIL import Image
from pi_heif import register_heif_opener

register_heif_opener()

im = Image.open("images/input.heic")  # do whatever need with a Pillow image
im.show()
```

So only `pi-heif` is required and two changes in the code:
```python
from pi_heif import register_heif_opener
register_heif_opener()
```

**Comment by tttdevelop** [comment]
Thanks for the tip. This fell off my radar a bit so thanks also @Vincent-Stragier for shaking this up. I'll update with the pi-heif.

**Comment by serengil** [comment]
closed due to inactivity

**Pull request closed without merging**